common/test/acceptance/tests/lms/test_lms_problems.py

```diff
@@ -1087,3 +1088,114 @@ def test_reset_correctness_after_changing_answer(self, input_value, correctness,
+        self.courseware_page = CoursewarePage(self.browser, self.course_id)
+
+        # Install a course with a hierarchy and problems
+        self.course_fixture = CourseFixture(
+            self.course_info['org'], self.course_info['number'],
+            self.course_info['run'], self.course_info['display_name'],
+            start_date=datetime.now() + timedelta(days=10)
+        )
+        self.browser.set_window_size(1024, 1024)
+
+    def setup_sequential(self, metadata):
+        """
+        Setup a sequential with DnD problem, alongwith the metadata provided.
+
+        This method will allow to customize the sequential, such as changing the
+        due date for individual tests.
+        """
+        problem = self.get_problem()
+        sequential = self.get_sequential(metadata=metadata)
+        self.course_fixture.add_children(
+            XBlockFixtureDesc('chapter', 'Test Section').add_children(
+                sequential.add_children(problem)
+            )
+        ).install()
+
+        # Auto-auth register for the course.
+        AutoAuthPage(
+            self.browser,
+            username=self.username,
+            email=self.email,
+            password=self.password,
+            course_id=self.course_id,
+            staff=True
+        ).visit()
+
+    def format_date(self, date_value):
+        """
+        Get the date in isoformat as this is required format to add date data
+        in the sequential.
+        """
+        return date_value.isoformat()
+
+    def get_problem(self):
+        """
+        Creating a DnD problem with assessment mode
+        """
+        return XBlockFixtureDesc('drag-and-drop-v2', 'DnD', metadata={'mode': "assessment"})
+
+    def get_sequential(self, metadata=None):
+        return XBlockFixtureDesc('sequential', 'Test Subsection', metadata=metadata)
+
+    @ddt.data(
+        (datetime.now(), True),
+        (datetime.now() - timedelta(days=1), True),
+        (datetime.now() + timedelta(days=1), False)
+    )
+    @ddt.unpack
+    def test_submit_button_status_with_due_date(self, due_date, is_button_disabled):
+        """
+        Scenario: Test that DnD submit button will be enabled if section is not past due.
+
+        Given I have a sequential in instructor-paced course
+        And a DnD problem with assessment mode is present in the sequential
+        When I visit the problem
+        Then the submit button should be present
+        And button should be disabled as some item needs to be on a zone
+        When I drag an item to a zone
+        Then submit button will be enabled if due date has not passed, else disabled
+        """
+        problem_page = DragAndDropPage(self.browser)
+        self.setup_sequential(metadata={'due': self.format_date(due_date)})
+        self.courseware_page.visit()
+        self.assertTrue(problem_page.is_submit_button_present())
+        self.assertTrue(problem_page.is_submit_disabled())
```

Comment:
why the submit button is disabled => `True` for due date `datetime.now() + timedelta(days=1)`?

Reply:
Initially, when no item has been dragged onto any zone, the submit button will remain disabled. Think it like other problems, like multiple choice. We can't submit unless we select an option.

Reply:
Sounds good.